### PR TITLE
Update server deps

### PR DESCRIPTION
**Pull request by Zolmeister**
depends on #160 

Last fix. If the issue persists after this, we'll have to go much deeper.

### CODE REVIEW

File: server.coffee [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-url = require 'url'
 _ = require 'lodash'
 Promise = require 'bluebird'
-request = Promise.promisifyAll require 'request'
 useragent = require 'express-useragent'
 compress = require 'compression'
 log = require 'loglevel'
 cookieParser = require 'cookie-parser'
-rp = require 'request-promise'
+request = require 'request-promise'
```
Comment (austinhallock) [inline server.coffee]: thanks

File: server.coffee [hunk 2]
```diff
@@ -58,9 +56,9 @@ clayUserSessionMiddleware = ->
       if isSubdomain
         return next()
 
-      request.postAsync loginUrl
+      Promise.cast request.post loginUrl
```
Comment (austinhallock) [inline server.coffee]: Why do we need to cast?

Comment (Zolmeister) [inline server.coffee]: request-promise only exposes `.then` from bluebird

Comment (Zolmeister) [inline server.coffee]: might be worth  forking and having it return a full bluebird promise

Comment (austinhallock) [inline server.coffee]: this is fine for now

**Pull request merged**